src/autogluon/bench/eval/hardware_metrics/hardware_metrics.py

```diff
@@ -0,0 +1,247 @@
+                utc_dt - timedelta(minutes=predict_time),
+            )
+            predict_util = get_instance_util(
+                instance_id, f"{metric}", utc_dt - timedelta(minutes=predict_time), utc_dt
+            )
+            # print(training_util, predict_util)
+            if training_util["Datapoints"]:
+                metrics_list.append(format_metrics(training_util, framework, dataset, fold, "Training"))
+            if predict_util["Datapoints"]:
+                metrics_list.append(format_metrics(predict_util, framework, dataset, fold, "Prediction"))
+
+
+def results_to_csv():
+    """
+    Writes the formatted dictionary of metrics to a csv to pass into `autogluon-dashboard`.
+    """
+    csv_headers = ["framework", "dataset", "mode", "fold", "metric", "statistic_type", "statistic_value", "unit"]
+    file_dir = os.path.dirname(__file__)
+    csv_location = os.path.join(file_dir, "hardware_metrics.csv")
+    with open(csv_location, "w", newline="") as csvFile:
+        writer = csv.DictWriter(csvFile, fieldnames=csv_headers)
+        writer.writeheader()
+        writer.writerows(metrics_list)
+
+
+def get_hardware_metrics(
+    config_file: str = typer.Argument(help="Path to YAML config file containing job ids."),
+    s3_bucket: str = typer.Argument(help="Name of the S3 bucket to which the benchmark results were outputted."),
+    module: str = typer.Argument(help="Can be one of ['tabular', 'multimodal']."),
+    benchmark_name: str = typer.Argument(
+        help="Folder name of benchmark run in which all objects with path 'scores/results.csv' get aggregated."
+    ),
+):
+    """
+    External API function to interact with the script.
+    Parameters
+    ----------
+    config_file: str,
+        Path to config file containing job IDs
+    s3_bucket: str,
+        Name of the S3 bucket to which the benchmark results were outputted.
+    module: str,
+        Benchmark module: tabular or multimodal
+    benchmark_name: str,
+        Name of the benchmark
+        Example: ag_bench_20230817T123456
+    """
+    if not config_file:
+        raise ValueError("Invalid Config File")
+    logger.info(f"Getting hardware metrics for jobs under config file: {config_file}")
+    with open(config_file, "r") as f:
+        config = yaml.safe_load(f)
+    job_ids = get_job_ids(config)
+
+    global metrics_list, aws_account_id, aws_account_region
+    aws_account_id = config.get("CDK_DEPLOY_ACCOUNT")
+    aws_account_region = config.get("CDK_DEPLOY_REGION")
+    for job_id in job_ids:
+        sub_folder = config["job_configs"][f"{job_id}"].split("/")[5].replace("_split", "").replace(".yaml", "")
+        get_metrics(
+            job_id, ["CPUUtilization", "EBSWriteOps", "EBSReadOps"], s3_bucket, module, benchmark_name, sub_folder
+        )
+    results_to_csv()
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+if __name__ == "__main__":
+    typer.run(get_hardware_metrics)
```

Comment:
I think with `typer`, we can directly run `agbench get-hardware-metrics`. So maybe this is unnecessary?

Reply:
I only added this if we wanted the ability to run this script independently in the CI